From 4068393acd75a415cc7dc532ccad4abaa9e27eb3 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Sun, 4 Jun 2023 18:21:07 +0200
Subject: [PATCH 3/5] meta-isar: jh7110-u-boot-spl-tool: Rework

Setting PACKAGE_ARCH is no longer needed, we can ask via "-native" for
the desired build type. Furthermore, install the binary as normal tool
under /usr/bin to simplify its usage.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 meta-isar/recipes-bsp/jh7110-u-boot-spl-image/files/rules    | 2 +-
 .../jh7110-u-boot-spl-image/jh7110-u-boot-spl-image_0.1.bb   | 2 +-
 .../jh7110-u-boot-spl-tool/jh7110-u-boot-spl-tool_0.1.bb     | 5 +----
 3 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/meta-isar/recipes-bsp/jh7110-u-boot-spl-image/files/rules b/meta-isar/recipes-bsp/jh7110-u-boot-spl-image/files/rules
index 6a25dd36..a8b2c6b8 100644
--- a/meta-isar/recipes-bsp/jh7110-u-boot-spl-image/files/rules
+++ b/meta-isar/recipes-bsp/jh7110-u-boot-spl-image/files/rules
@@ -4,7 +4,7 @@ U_BOOT_SPL_BIN := /usr/lib/u-boot/starfive-visionfive2/u-boot-spl.bin
 
 override_dh_auto_build:
 	cp $(U_BOOT_SPL_BIN) u-boot-spl.bin
-	/usr/lib/jh7110-uboot-spl-tool/spl_tool -c -f u-boot-spl.bin
+	spl_tool -c -f u-boot-spl.bin
 
 %:
 	dh \$@
diff --git a/meta-isar/recipes-bsp/jh7110-u-boot-spl-image/jh7110-u-boot-spl-image_0.1.bb b/meta-isar/recipes-bsp/jh7110-u-boot-spl-image/jh7110-u-boot-spl-image_0.1.bb
index 08c53b51..f237a9c3 100644
--- a/meta-isar/recipes-bsp/jh7110-u-boot-spl-image/jh7110-u-boot-spl-image_0.1.bb
+++ b/meta-isar/recipes-bsp/jh7110-u-boot-spl-image/jh7110-u-boot-spl-image_0.1.bb
@@ -8,7 +8,7 @@ inherit dpkg
 SRC_URI = "file://rules file://install"
 
 DESCRIPTION = "StarFive VisionFive 2 u-boot SPL"
-DEPENDS = "jh7110-u-boot-spl-tool u-boot-starfive-visionfive2"
+DEPENDS = "jh7110-u-boot-spl-tool-native u-boot-starfive-visionfive2"
 DEBIAN_BUILD_DEPENDS = "jh7110-u-boot-spl-tool, u-boot-starfive-visionfive2:${DISTRO_ARCH}"
 
 # this is a host tool
diff --git a/meta-isar/recipes-bsp/jh7110-u-boot-spl-tool/jh7110-u-boot-spl-tool_0.1.bb b/meta-isar/recipes-bsp/jh7110-u-boot-spl-tool/jh7110-u-boot-spl-tool_0.1.bb
index 1835d474..42d3b9d8 100644
--- a/meta-isar/recipes-bsp/jh7110-u-boot-spl-tool/jh7110-u-boot-spl-tool_0.1.bb
+++ b/meta-isar/recipes-bsp/jh7110-u-boot-spl-tool/jh7110-u-boot-spl-tool_0.1.bb
@@ -13,11 +13,8 @@ SRCREV = "8c5acc4e5eb7e4ad012463b05a5e3dbbfed1c38d"
 
 S = "${WORKDIR}/tools/spl_tool"
 
-# This is a host tool
-PACKAGE_ARCH = "${HOST_ARCH}"
-
 do_prepare_build[cleandirs] += "${S}/debian"
 do_prepare_build(){
     deb_debianize
-    echo "spl_tool usr/lib/jh7110-uboot-spl-tool" > ${S}/debian/${PN}.install
+    echo "spl_tool usr/bin" > ${S}/debian/${BPN}.install
 }
-- 
2.34.1

